Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -811,30 +838,31 @@ void FrameLoader::didBeginDocument(bool dispatch)
 
 void FrameLoader::finishedParsing()
 {
-    LOG(Loading, "WebCoreLoading frame %" PRIu64 ": Finished parsing", m_frame.frameID().object().toUInt64());
-
-    m_frame.injectUserScripts(UserScriptInjectionTime::DocumentEnd);
-
-    if (m_stateMachine.creatingInitialEmptyDocument())
-        return;
+    LOG(Loading, "WebCoreLoading frame %" PRIu64 ": Finished parsing", m_frame->frameID().object().toUInt64());
 
     // This can be called from the Frame's destructor, in which case we shouldn't protect ourselves
     // because doing so will cause us to re-enter the destructor when protector goes out of scope.
     // Null-checking the FrameView indicates whether or not we're in the destructor.
-    RefPtr protectedFrame { m_frame.view() ? &m_frame : nullptr };
+    RefPtr protectedFrame { m_frame->view() ? m_frame.ptr() : nullptr };
+
+    m_frame->injectUserScripts(UserScriptInjectionTime::DocumentEnd);
```

Comment:
Can we used protectedFrame here?

Reply:
protectedFrame can be null, see 2 lines above. This function may get called during the frame destruction.